sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -74,12 +75,26 @@ class QueryExecution(
     sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker)
   }
 
+  lazy val nonRootCommandExecuted: LogicalPlan = analyzed mapChildren { child =>
+    child transformDown {
+      // SPARK-35378: Eagerly execute non-root Command
+      case c: Command =>
+        val qe = sparkSession.sessionState.executePlan(c)
+        CommandResult(
+          qe.nonRootCommandExecuted.output,
+          qe.nonRootCommandExecuted,
```

Comment:
Can `Command` contain other `Command`? Otherwise, can we just use `qe.analyzed.output` here?